src/misc/rwqueue.cpp

```diff
@@ -181,6 +193,29 @@ bool RWQueue<T>::BulkEnqueue(std::vector<T>& from_source, const size_t num_reque
 	return is_running;
 }
 
+template <typename T>
+bool RWQueue<T>::NonblockingBulkEnqueue(std::vector<T>& from_source, const size_t num_requested)
+{
+	assert(num_requested > 0);
+	assert(num_requested <= from_source.size());
+
+	std::lock_guard<std::mutex> lock(mutex);
+	if (!is_running || queue.size() >= capacity) {
+		return false;
```

Comment:
> queue.size() >= capacity

How can the `>` condition even happen? Or would `=` be enough and `>=` is just for extra "safety"?

Reply:
The capacity can be shrunk at runtime.  Changing the capacity does not change the size.  It just sets a value that prevents more elements from being added.

Reply:
Okay, that's new to me 😅 I did not think that was possible.